tools/src/main/java/org/apache/kafka/trogdor/task/TaskSpec.java

```diff
@@ -74,6 +75,18 @@ public final long durationMs() {
         return durationMs;
     }
 
+    /**
+     * Whether the task is considered expired
+     * @param startedMs the actual time that this task started on
+     */
+    public boolean hasExpired(Time time, long startedMs) {
+        long startMs = this.startMs > 0 ? this.startMs : startedMs;
+        if (startMs <= 0) // task doesn't have a start time yet
+            return false;
```

Comment:
We shouldn't be special-casing 0 here.  In general, I think the only thing we need here is an accessor function like `endMs` which returns `startMs + durationMs`.